addons/medical/functions/fnc_actionRemoveTourniquet.sqf

```diff
@@ -34,3 +35,20 @@ _target setVariable [QGVAR(tourniquets), _tourniquets, true];
 
 // Adding the tourniquet item to the caller
 _caller addItem "ACE_tourniquet";
+
+//Handle all injected medications now that blood is flowing:
+private _delayedMedications = _target getVariable [QGVAR(occludedMedications), []];
+private _removed = [];
+TRACE_2("meds",_part,_delayedMedications);
+{
+    _x params ["", "", "_medPartNum"];
+    if (_part == _medPartNum) then {
+        TRACE_1("delayed medication call after tourniquet removeal",_x);
+        ["medical_advMedication", [_target], _x] call EFUNC(common,targetEvent);
+        _removed pushBack _x;
```

Comment:
Why not replace the value by `objNull` and delete those after the for each? I'm not sure i Ilike the array for removal.